copy/src/redux/types/index.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    SET_LOADING,
+    SET_QUESTIONS,
+    SET_USER_ANSWERS,
+    SET_SCORE,
+    SET_CURRENT_NUMBER,
+    SET_GAME_OVER
+} from '../actions'
+
+
+export enum Difficulty {
+    Easy = 'easy',
+    Medium = 'medium',
+    Hard = 'hard'
+};
+
+export interface AnswerObject {
+    question: string;
+    userAnswer: string;
+    isCorrect: boolean;
+    correctAnswer: string;
+};
+
+export interface GameState {
+    loading: boolean,
+    questions: QuestionState[],
+    userAnswers: AnswerObject[],
+    score: number,
+    currentNumber: number,
+    gameOver: boolean,
+}
+
+export interface Question {
+    category: string,
+    correct_answer: string,
+    difficulty: string,
+    incorrect_answers: string[],
+    question: string,
+    type: string,
+}
+
+//ACTION TYPES
+
+export interface setLoading {
+    type: typeof SET_LOADING,
+    payload: boolean;
+}
+
+export interface setQuestions {
+    type: typeof SET_QUESTIONS,
+    payload: QuestionState[],
+}
+
+export interface setUserAnswers {
+    type: typeof SET_USER_ANSWERS,
+    payload: AnswerObject[],
+}
+
+export interface setScore {
+    type: typeof SET_SCORE,
+    payload: number,
+}
+
+export interface setCurrentNumber {
+    type: typeof SET_CURRENT_NUMBER,
+    payload: number,
+}
+
+export interface setGameOver {
+    type: typeof SET_GAME_OVER,
+    payload: boolean,
+}
+
+
+
+export type QuestionState = Question & { answers: string[] };
```

Comment:
I think you are not using `Question` type by itself anywhere, so `QuestionState` and `Question` should be merged into one type.

Also, even if you did need two separate ones, QuestionsState should be an interface that extends Question.
As a rule of thumb, if it's an object or defining object-like properties, it should be an interface

types are more for basic types and using "or" on multiple types (like you did with GameStateActionTypes

Coming back to naming `GameStateActionTypes` are better named as `GameStateActions`